### DISCUSSION TITLE
Tuning ROSCO without FAST Files

### DISCUSSION THREAD

**Discussion by MaximilienAndre** [NatLabRockies/ROSCO#339] [Q&A] [answered]
Dear all, 

I am fearly new to ROSCO and as I move through the tutorials there is something I would like to understand. 

When running this lines (exemple 03 ) 
```

# Instantiate turbine, controller, and file processing classes
turbine         = ROSCO_turbine.Turbine(turbine_params)
controller      = ROSCO_controller.Controller(controller_params)

# Load turbine data from OpenFAST and rotor performance text file
cp_filename = os.path.join(tune_dir,path_params['rotor_performance_filename'])
turbine.load_from_fast(
    path_params['FAST_InputFile'],
    os.path.join(tune_dir,path_params['FAST_directory']),
    rot_source='txt',txt_filename= cp_filename
    )
```

I understand that we are defining a turbine object, that will be used so that the controler is appropriately tuned for this turbine. What I don't understand is what parameters are necessary from the turbine object. Apparently the parameters from the YAML file are not enough because if I comment the `turbine.load_from_fast` line the tuning function returns errors. 

So if I understand correctly, we are loading parameters both from the YAML file and from the FAST Files. So my question is what parameters from the fast files are being used ? A subsidiary question is, let's say I have a new turbine model without associated fast files. How can I tune ROSCO for this turbine without having to develop a full OpenFAST model for it ? 

Best, 

Maximilien

**Comment by dzalkind** [comment, marked as answer]
Hi Maximilien,

Sorry for the delay.

I recommend reviewing the ROSCO `turbine` class.  Here is where the OpenFAST information is read: https://github.com/NREL/ROSCO/blob/7deb583d4acbacc068bad28e2228f7d257a7cd7c/rosco/toolbox/turbine.py#L203

I would try populating the attributes of the ROSCO `turbine` object with the information gleaned from OpenFAST in another way.  Here is an example: https://github.com/WISDEM/WEIS/blob/394d9223dea784bac68eca7fc9be9f846516f841/weis/control/tune_rosco.py#L230 

Best, Dan

**Reply by MaximilienAndre** [reply]
Dear Dan, 

Thank you for your answer, I will look at the ROSCO turbine class and see if I can set all the parameters myself. 

Best, Maximilien

**Reply by nipscowi** [reply]
Hi Maximillien
I am looking into the same topic I believe and would like to hear if you succeeded in tuning the ROSCO controller without an OpenFAST model?

In my case I have a Bladed model, that I would like to run with the ROSCO controller. So how can I set up the ROSCO controller tuning without an OpenFAST model?

I imagine I can run rotor performance simulations in Bladed to prepare a Cp-surface data set for for the controller tuning. But how do I replace the OpenFAST model or the input taken from it?

Do you have any advise?
